library/core/tests/const_ptr.rs

```diff
@@ -0,0 +1,51 @@
+// Aligned to two bytes
+const DATA: [u16; 2] = [u16::from_ne_bytes([0x01, 0x23]), u16::from_ne_bytes([0x45, 0x67])];
+
+const fn unaligned_ptr() -> *const u16 {
+    // Since DATA.as_ptr() is aligned to two bytes, adding 1 byte to that produces an unaligned *const u16
+    unsafe { (DATA.as_ptr() as *const u8).add(1) as *const u16 }
+}
+
+#[test]
+fn read() {
+    use core::ptr;
+
+    const FOO: i32 = unsafe { ptr::read(&42 as *const i32) };
+    assert_eq!(FOO, 42);
+
+    const ALIGNED: i32 = unsafe { ptr::read_unaligned(&42 as *const i32) };
+    assert_eq!(ALIGNED, 42);
+
+    const UNALIGNED_PTR: *const u16 = unaligned_ptr();
+
+    const UNALIGNED: u16 = unsafe { ptr::read_unaligned(UNALIGNED_PTR) };
+    assert_eq!(UNALIGNED, u16::from_ne_bytes([0x23, 0x45]));
```

Comment:
Is the unaligned test safe and valid? Is there any better way to test an unaligned read?

Reply:
Also, will this work on both big and little endian systems?

Reply:
> Is the unaligned test safe and valid? Is there any better way to test an unaligned read?

The test looks great.

> Also, will this work on both big and little endian systems?

Good question. I think you can avoid this by using `from_le/be_bytes` -- if you avoid `ne` ("native endian"), the host endianess should not matter, right?

Reply:
I might be completely wrong but would not `let x = u16::from_le_bytes([0x01, 0x00])` just ensure that `x == 1` on all platforms?

It is my believe that we, in this case, do not care whether `x` is `1` or `256` as long as we know what byte ends up at what address. So I guess the question would be does my code do that?

Reply:
> I might be completely wrong but would not let x = u16::from_le_bytes([0x01, 0x00]) just ensure that x == 1 on all platforms?

Hm, I guess you are right.

I have never worked with different endianesses (is that a word?^^), so I don't have good intuition here.

Reply:
> So I guess the question would be does my code do that?

I guess CI will tell us when we merge this PR. :D